<a href="https://colab.research.google.com/github/MateusSanfer/Desafio-de-codigo-AluraStore-Br/blob/main/AluraStoreBr.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importação dos dados



In [44]:
import pandas as pd
import matplotlib.pyplot as plt

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science/refs/heads/main/base-de-dados-challenge-1/loja_1.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science/refs/heads/main/base-de-dados-challenge-1/loja_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science/refs/heads/main/base-de-dados-challenge-1/loja_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science/refs/heads/main/base-de-dados-challenge-1/loja_4.csv"

loja = pd.read_csv(url)
loja2 = pd.read_csv(url2)
loja3 = pd.read_csv(url3)
loja4 = pd.read_csv(url4)

loja.head()

Produto Categoria do Produto    Preço      Frete  \
0     Assistente virtual          eletronicos   219.08   9.249790   
1         Mesa de jantar               moveis   256.35  11.234305   
2      Jogo de tabuleiro           brinquedos   279.51  21.262681   
3            Micro-ondas     eletrodomesticos  1009.99  54.667344   
4  Cadeira de escritório               moveis   446.99  26.964689   

  Data da Compra        Vendedor Local da compra  Avaliação da compra  \
0     16/01/2021     Pedro Gomes              SP                    4   
1     18/05/2022  Beatriz Moraes              RJ                    1   
2     15/03/2021      João Souza              DF                    1   
3     03/05/2022      João Souza              RS                    4   
4     07/11/2020   Larissa Alves              MG                    5   

  Tipo de pagamento  Quantidade de parcelas    lat    lon  
0    cartao_credito                       8 -22.19 -48.79  
1    cartao_credito                       4 -22.25 -42.66  
2    cartao_credito                       1 -15.83 -47.86  
3            boleto                       1 -30.17 -53.50  
4            boleto                       1 -18.10 -44.38

#1. Análise do faturamento


In [48]:
print('Análise do faturamento: ')
faturamento ={
    "Loja 01": loja["Preço"].sum(),
    "Loja 02": loja2["Preço"].sum(),
    "Loja 03": loja3["Preço"].sum(),
    "Loja 04": loja4["Preço"].sum(),
}
for nome_lojas, valor in faturamento.items():
    print(f'{nome_lojas}: R${valor:,.2f}')

Análise do faturamento: 
Loja 01: R$1,534,509.12
Loja 02: R$1,488,459.06
Loja 03: R$1,464,025.03
Loja 04: R$1,384,497.58


# 2. Vendas por Categoria


In [49]:
print('Vendas por categoria: ')
venda_por_categoria = {
    "Loja 01": loja["Categoria do Produto"].value_counts(),
    "Loja 02": loja2["Categoria do Produto"].value_counts(),
    "Loja 03": loja3["Categoria do Produto"].value_counts(),
    "Loja 04": loja4["Categoria do Produto"].value_counts(),
}

for lojas, categorias in venda_por_categoria.items():
    print(f"Loja: {lojas}")
    print(categorias)
    print()


Vendas por categoria: 
Loja: Loja 01
Categoria do Produto
moveis                   465
eletronicos              448
brinquedos               324
eletrodomesticos         312
esporte e lazer          284
instrumentos musicais    182
livros                   173
utilidades domesticas    171
Name: count, dtype: int64

Loja: Loja 02
Categoria do Produto
moveis                   442
eletronicos              422
brinquedos               313
eletrodomesticos         305
esporte e lazer          275
instrumentos musicais    224
livros                   197
utilidades domesticas    181
Name: count, dtype: int64

Loja: Loja 03
Categoria do Produto
moveis                   499
eletronicos              451
brinquedos               315
eletrodomesticos         278
esporte e lazer          277
livros                   185
instrumentos musicais    177
utilidades domesticas    177
Name: count, dtype: int64

Loja: Loja 04
Categoria do Produto
moveis                   480
eletronicos              451
br

# 3. Média de Avaliação das Lojas

In [41]:
print('Média de Avaliação das Lojas: ')
media_avaliacao_loja = {
    "Loja 01": loja["Avaliação da compra"].mean(),
    "Loja 02": loja2["Avaliação da compra"].mean(),
    "Loja 03": loja3["Avaliação da compra"].mean(),
    "Loja 04": loja4["Avaliação da compra"].mean(),
}

for lojas, media_nota in media_avaliacao_loja.items():
  print(f'{lojas}:  {media_nota:.2f}')



Média de Avaliação das Lojas: 
Loja 01:  3.98
Loja 02:  4.04
Loja 03:  4.05
Loja 04:  4.00


# 4. Produtos Mais e Menos Vendidos

In [47]:
print('Produtos Mais e Menos Vendidos: ')
produtos_vendidos = {
    "Loja 01": loja["Produto"].value_counts(),
    "Loja 02": loja2["Produto"].value_counts(),
    "Loja 03": loja3["Produto"].value_counts(),
    "Loja 04": loja4["Produto"].value_counts(),
}

for loja_nome, produtos in produtos_vendidos.items():
  produto_mais_vendido = produtos.idxmax()
  quantidade_mais_vendido = produtos.max()

  produto_menos_vendido = produtos.idxmin()
  quantidade_menos_vendido = produtos.min()

  print(f"\n{loja_nome}:")
  print(f"Produto mais vendido: {produto_mais_vendido} (Vendas: {quantidade_mais_vendido})")
  print(f"Produto menos vendido: {produto_menos_vendido} (Vendas: {quantidade_menos_vendido})")

Produtos Mais e Menos Vendidos: 

Loja 01:
Produto mais vendido: Micro-ondas (Vendas: 60)
Produto menos vendido: Headset (Vendas: 33)

Loja 02:
Produto mais vendido: Iniciando em programação (Vendas: 65)
Produto menos vendido: Jogo de tabuleiro (Vendas: 32)

Loja 03:
Produto mais vendido: Kit banquetas (Vendas: 57)
Produto menos vendido: Blocos de montar (Vendas: 35)

Loja 04:
Produto mais vendido: Cama box (Vendas: 62)
Produto menos vendido: Guitarra (Vendas: 33)


# 5. Frete Médio por Loja


In [52]:
print('Frete Médio por Loja: ')
frete_medio= {
    "Loja 01": loja["Frete"].mean(),
    "Loja 02": loja2["Frete"].mean(),
    "Loja 03": loja3["Frete"].mean(),
    "Loja 04": loja4["Frete"].mean(),
}

for nome_lojas, valor_frete in frete_medio.items():
  print(f"{nome_lojas}: R${valor_frete:,.2f}")

Frete Médio por Loja: 
Loja 01: R$34.69
Loja 02: R$33.62
Loja 03: R$33.07
Loja 04: R$31.28


## Relatorio

### **Relatório de Análise de Vendas e Desempenho das Lojas - Alura Store**

#### **1. Introdução**

O objetivo deste relatório é fornecer uma análise detalhada sobre o desempenho das lojas da rede Alura Store, ajudando o Senhor João a decidir qual loja deve ser vendida para iniciar um novo empreendimento. A análise baseia-se em métricas de **faturamento**, **avaliação de clientes**, **vendas por categoria**, **produtos mais e menos vendidos** e **frete médio**.

#### **2. Análise de Faturamento**

A primeira métrica analisada foi o **faturamento** total de cada loja. O faturamento é um dos indicadores mais importantes, pois demonstra a capacidade de geração de receita de cada unidade.

- **Loja 01**: R$1.534.509,12
- **Loja 02**: R$1.488.459,06
- **Loja 03**: R$1.464.025,03
- **Loja 04**: R$1.384.497,58

A **Loja 01** apresentou o maior faturamento, enquanto a **Loja 04** ficou com o menor valor, destacando-se como uma loja com **menor desempenho financeiro**.

#### **3. Análise de Vendas por Categoria**

Em relação às **vendas por categoria**, as lojas possuem um desempenho muito semelhante, com destaque para as categorias de **móveis** e **eletrônicos**, que são as mais vendidas em todas as lojas.

- **Loja 01**: 465 unidades de móveis, 448 de eletrônicos.
- **Loja 02**: 442 unidades de móveis, 422 de eletrônicos.
- **Loja 03**: 499 unidades de móveis, 451 de eletrônicos.
- **Loja 04**: 480 unidades de móveis, 451 de eletrônicos.

Embora a diferença não seja grande, a **Loja 04** tem números **ligeiramente menores** em comparação com as demais, indicando um desempenho um pouco mais fraco.

#### **4. Análise das Avaliações**

As **médias de avaliação** também foram analisadas. A avaliação dos clientes reflete diretamente a satisfação com os produtos e serviços de cada loja:

- **Loja 01**: 3,98
- **Loja 02**: 4,04
- **Loja 03**: 4,05
- **Loja 04**: 4,00

A **Loja 03** obteve a maior média de avaliação, destacando-se em termos de **satisfação do cliente**. A **Loja 01** teve a menor média, mas a diferença não é tão significativa.

#### **5. Produtos Mais e Menos Vendidos**

A análise de **produtos mais e menos vendidos** mostra os seguintes resultados:

- **Loja 01**: Produto mais vendido: Micro-ondas (60 vendas), produto menos vendido: Headset (33 vendas).
- **Loja 02**: Produto mais vendido: Iniciando em programação (65 vendas), produto menos vendido: Jogo de tabuleiro (32 vendas).
- **Loja 03**: Produto mais vendido: Kit banquetas (57 vendas), produto menos vendido: Blocos de montar (35 vendas).
- **Loja 04**: Produto mais vendido: Cama box (62 vendas), produto menos vendido: Guitarra (33 vendas).

Os produtos mais vendidos variam entre **57 e 65 unidades**, enquanto os menos vendidos estão entre **32 e 35 unidades**. Isso indica que todas as lojas têm um **desempenho equilibrado** no que diz respeito aos produtos populares, mas com pequenas variações.

#### **6. Frete Médio**

O **frete médio** é uma métrica importante para entender o custo logístico das lojas. Os resultados obtidos foram:

- **Loja 01**: R$34,69
- **Loja 02**: R$33,62
- **Loja 03**: R$33,07
- **Loja 04**: R$31,28

A **Loja 04** tem o **menor custo de frete**, o que pode ser um ponto positivo, mas não é suficiente para compensar seu desempenho inferior em outras métricas.

#### **7. Conclusão e Recomendação**

Com base nas análises de **faturamento**, **vendas por categoria**, **avaliações** dos clientes, **produtos mais e menos vendidos** e **frete médio**, concluímos que a **Loja 04** apresenta o **menor desempenho geral** entre as quatro.

Embora o custo de frete da Loja 04 seja o menor, seu **faturamento** e **vendas por categoria** são inferiores às outras lojas. Além disso, a **média de avaliação** dos clientes e os **produtos vendidos** também indicam um desempenho mais fraco.

Portanto, **recomendamos que o Senhor João considere vender a Loja 04**, pois ela se destaca como a opção menos eficiente, com menores resultados financeiros e de desempenho em relação às outras lojas.
